auth.php

```diff
@@ -53,14 +53,15 @@
     // Set state from user actions.
     if ($form->is_cancelled()) {
         $factor->set_state(\tool_mfa\plugininfo\factor::STATE_NEUTRAL);
+        // Move to next factor.
+        \tool_mfa\manager::check_status(true);
     } else {
         if ($data = $form->get_data()) {
             $factor->set_state(\tool_mfa\plugininfo\factor::STATE_PASS);
+            // Move to next factor.
+            \tool_mfa\manager::check_status(true);
```

Comment:
could these 2 calls just be one and moved outside the ifs?

Reply:
Yeah This can be structured better

Reply:
ah I saw it was like that but then moved here. So curious why?

Reply:
This was causing an issue where the page would attempt to redirect instead of displaying the validation. It still allowed good requests to pass, but caused display issues